crates/stellar-scaffold-cli/src/commands/generate/contract/mod.rs

```diff
@@ -112,14 +77,134 @@ impl Cmd {
+        let workspace = manifest.workspace.as_mut().unwrap();
+
+        let mut workspace_deps = workspace.dependencies.clone();
+
+        let mut added_deps = Vec::new();
+        let mut updated_deps = Vec::new();
+
+        for dep in deps {
+            let git_dep = cargo_toml::DependencyDetail {
+                git: Some("https://github.com/OpenZeppelin/stellar-contracts".to_string()),
+                tag: Some(tag.to_string()),
+                ..Default::default()
+            };
+
+            if let Some(existing_dep) = workspace_deps.clone().get(&dep) {
+                // Check if we need to update the tag
+                if let Detailed(DependencyDetail { tag: Some(existing_tag), ..}) = &existing_dep && existing_tag != tag {
+                      workspace_deps.insert(
+                          dep.clone(),
+                          cargo_toml::Dependency::Detailed(Box::new(git_dep)),
+                      );
+                      updated_deps.push((dep, existing_tag.clone()));
+                }
+            } else {
+                workspace_deps.insert(
+                    dep.clone(),
+                    cargo_toml::Dependency::Detailed(Box::new(git_dep)),
+                );
+                added_deps.push(dep);
+            }
+        }
+
+        if !added_deps.is_empty() || !updated_deps.is_empty() {
+            workspace.dependencies = workspace_deps;
+            // Write the updated manifest back to file
+            let toml_string = toml::to_string_pretty(&manifest)?;
+            fs::write(workspace_path, toml_string)?;
+
+            if !added_deps.is_empty() {
+                eprintln!("📦 Added the following dependencies to workspace:");
+                for dep in added_deps {
+                    eprintln!("   • {dep}");
+                }
+            }
+
+            if !updated_deps.is_empty() {
+                eprintln!("🔄 Updated the following dependencies:");
+                for (dep, old_tag) in updated_deps {
+                    eprintln!("   • {dep}: {old_tag} -> {tag}");
+                }
+            }
+        }
+
+        Ok(())
+    }
+
+    fn extract_stellar_dependencies(cargo_toml_content: &str) -> Result<Vec<String>, Error> {
+        let manifest: cargo_toml::Manifest = toml::from_str(cargo_toml_content)?;
+        let mut deps = Vec::new();
+
+        for (dep_name, dep_detail) in &manifest.dependencies {
+            if dep_name.starts_with("stellar-") {
+                match dep_detail {
+                    cargo_toml::Dependency::Inherited(_) | cargo_toml::Dependency::Simple(_) => {
+                        deps.push(dep_name.clone());
+                    }
+                    cargo_toml::Dependency::Detailed(detail) => {
+                        if detail.inherited || detail.git.is_some() {
+                            deps.push(dep_name.clone());
+                        }
+                    }
+                }
+            }
+        }
+
+        Ok(deps)
```

Comment:
I might go over board with my distaste for for loops + mutable vector pushes. But IMO it reads better to use iterators and make it clearer what the goals are.

Also this is definitely not formatted correctly! Sorry!

```suggestion
        Ok(manifest.dependencies
              .iter()
             .filter(|(dep_name, _)| dep_name.starts_with("stellar-"))
             .filter_map((|(dep_name, dep_detail)| {
                if let Dependency::Detailed(detail) && !(detail.inherited || detail.git.is_some()) { 
                    None 
               } else {
                    Some(dep_name.clone())
              }).collect())
```